### ISSUE TITLE
Rancher goes bananas and take 99% CPU

### ISSUE THREAD

**Issue by mckn** [rancher/rancher#6030]
**Rancher Version:**
Rancher: v1.1.3
Cattle: v0.165.8.1

**Docker Version:**
1.12.1

**OS and where are the hosts located? (cloud, bare metal, etc):**
Ubuntu 16.04.1 LTS (4.4.0-38-generic)

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single node rancher with internal DB

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle

**Steps to Reproduce:**
We have a rancher server that we have been upgrading every release since 1.0. After the upgrade to 1.1.3 we have been required to restart the whole rancher server once a week. It looks like the JAVA process for the rancher catalogue service is taking close to 100%. 

I have attached the rancher server log: http://pastebin.com/CeSMrf0e

**Results:** 
The server stops responding after a while..

**Expected:**
The server should be running smooth and without any errors as before.

**Comment by aemneina** [comment]
@mckn the first few exceptions are sql related... looks like timeouts to the builtin db? Did the issues correlate in any way to a docker version upgrade? I dont believe 1.12 is supported at this point. Also what kind of spec is the host running the rancher container?

**Comment by elcct** [comment]
I experience the same behavior using the same configuration. Server stops responding after a while.
My nginx error log is full of:
```
2016/09/23 01:32:22 [error] 1139#1139: *52142 recv() failed (104: Connection reset by peer) while reading response header from upstream, client: [redacted], server: [redacted], request: "GET /v1/schemas/account HTTP/1.1", upstream: "http://127.0.0.1:8080/v1/schemas/account", host: "[redacted]"
```

I am not seeing any exceptions in the server log. 

Host is E3 1245 v5 with 64 GB ram.

**Comment by aemneina** [comment]
@elcct what does your nginx config look like? Please hide any sensitive info when sharing it.

**Comment by elcct** [comment]
```
server {                                                                                                                                                                
    listen [redacted] ssl;                                                                                                                                      
    server_name [redacted];                                                                                                                                       
    ssl_certificate /etc/ssl/private/xxx.crt;                                                                                                              
    ssl_certificate_key /etc/ssl/private/xxx.key;                                                                                                          
                                                                                                                                                                                                                                                                                                                                                
    error_log /var/log/nginx/[redacted].error.log;                                                                                                                          
    access_log /var/log/nginx/[redacted].access.log;                                                                                                                        
                                                                                                                                                                        
    location / {                                                                                                                                                        
        proxy_set_header Host $host;                                                                                                                                    
        proxy_set_header X-Forwarded-Proto $scheme;                                                                                                                     
        proxy_set_header X-Forwarded-Port $server_port;                                                                                                                 
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;                                                                                                    
        proxy_pass http://127.0.0.1:8080;                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
        proxy_http_version 1.1;                                                                                                                                         
        proxy_set_header Upgrade $http_upgrade;                                                                                                                         
        proxy_set_header Connection "Upgrade";                                                                                                                                                                                                                                                                                       
        proxy_read_timeout 900s;                                                                                                                                        
    }                                                                                                                                                                   
}                                                                                                                                                                     
```

Surprisingly I can do a `curl -i https://redacted`, it replies with:

```
HTTP/1.1 401 Unauthorized
Server: nginx/1.10.1
Date: Fri, 23 Sep 2016 18:34:28 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 158
Connection: keep-alive
Expires: Thu, 01 Jan 1970 00:00:00 GMT
Set-Cookie: PL=rancher;Path=/
Www-Authenticate: Basic realm="Enter API access key and secret key as username and password"
X-Api-Schemas: https://[redacted]/v1/schemas

{"id":"8a3bec7d-1d95-4ddc-83fd-9f9da90e4e01","type":"error","links":{},"actions":{},"status":401,"code":"Unauthorized","message":"Unauthorized","detail":null}
```                                                                                                                                                                  

But when I do it from another computer it times out. It doesn't seem to be a network problem, because when I replace `proxy_pass` with for example `root` and a path that has some file and restart nginx I can read that file from any host.

**Comment by aemneina** [comment]
I believe the proxy_pass should point to https://<backend_upstream_server_def> and not localhost.

**Comment by elcct** [comment]
@aemneina I found the problem - it was not the nginx configuration. Looks like I had `rancher/agent` running on one of the machines in my local network. 

- I added new host

- Installed agent on that host

- Deleted host

- Agent was still running on that host

Host where that agent was installed was sharing the same external ip as my dev pc. 

After I killed and deleted container with that agent from that host, Rancher UI started responding.
